### PR TITLE
[spark] Supports paimon_incremental_query Table Valued Function

### PR DESCRIPTION
**Pull request by YannByron**
- Builds Paimon SparkSession Extensions Arch;
- Supports paimon_incremental_query TVF to query the incr data between the two versions provided.

Linked issue: close #xxx

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Thanks! We can also modify documentation too.
In `/how-to/querying-tables`.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonTableValuedFunctions.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import org.apache.spark.sql.catalyst.plans.logical.{LeafNode, LogicalPlan}
+import org.apache.spark.sql.connector.catalog.Identifier
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+import scala.collection.JavaConverters._
+
+object PaimonTableValuedFunctions {
+
+  val INCREMENTAL_QUERY = "paimon_incremental_query"
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonTableValuedFunctions.scala:37]: Dose this need to be `paimon_` prefix? Dose `incremental_query` work?
How about Delta and Hudi?

Comment (YannByron) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonTableValuedFunctions.scala:37]: Delta doesn't use the prefix, but Hudi does. It's better to have, to distinguish when more then one SparkSessionExtensions are used.

File: docs/content/how-to/querying-tables.md [hunk 2]
```diff
@@ -129,7 +129,20 @@ SELECT * FROM t /*+ OPTIONS('incremental-between' = '12,20') */;
 ```
 {{< /tab >}}
 
-{{< tab "Spark" >}}
+{{< tab "Spark3" >}}
+
+Requires Spark 3.2+.
+
+you can use `paimon_incremental_query` in query to extract the incremental data:
```
Comment (JingsongLi) [inline docs/content/how-to/querying-tables.md:136]: Should also add Extension and use SparkGenericCatalog?

**Review by JingsongLi — APPROVED** [review]
+1 Thanks @YannByron

**Pull request merged**